src/Microsoft.TestPlatform.TestHostProvider/Hosting/DotnetTestHostManager.cs

```diff
@@ -259,6 +259,10 @@ public async Task<bool> LaunchTestHostAsync(TestProcessStartInfo testHostStartIn
             if (!testHostExeFound)
             {
                 var currentProcessPath = this.processHelper.GetCurrentProcessFileName();
+                if (testHostPath == string.Empty)
```

Comment:
Always string.IsNullOrEmpty

Reply:
How are you planning to test this ?

Reply:
Also think about converging the two calls, if not possible I am okay with as well

Reply:
Converted to string.IsNullOrEmpty.
Also added an extra assert in existing test to veryfy testhost.dll path is determined in non windows scenario.